File: auth-service/src/main/java/com/procurement/authservice/dto/LoginResponse.java
package com.procurement.authservice.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LoginResponse {
    private String accessToken;
    private String tokenType;
    private Long userId;
    private String email;
    private String fullName;
    /** Effective role for this session (may differ from procurement role in SALES context). */
    private String role;
    /** Primary procurement-side role stored on the user record. */
    private String procurementRole;
    private String supplierRole;
    private Long tenantId;
    private String tenantName;
    private String tenantDomain;
    private String organizationType;
    private String operatingContext;
    private List<String> availableContexts;
    private Boolean mustChangePassword;
}

File: frontend/lib/login-response.ts
import type { User } from "@/lib/auth-store";
import type { OperatingContext } from "@/lib/operating-context";
import type { OrganizationType } from "@/lib/operating-context";

export type LoginPayload = {
  accessToken: string;
  userId: number;
  email: string;
  fullName: string;
  role: string;
  procurementRole?: string;
  supplierRole?: string | null;
  tenantId: number;
  tenantName: string;
  tenantDomain: string;
  organizationType?: OrganizationType | string;
  operatingContext?: OperatingContext | string;
  availableContexts?: string[];
  mustChangePassword?: boolean;
};

export function mapLoginToUser(response: LoginPayload): User {
  const fullName = response.fullName || "";
  const nameParts = fullName.split(" ");
  return {
    id: String(response.userId || ""),
    userId: String(response.userId || ""),
    email: response.email || "",
    firstName: nameParts[0] || "",
    lastName: nameParts.slice(1).join(" ") || "",
    fullName,
    role: response.role as User["role"],
    roleName: response.role as User["roleName"],
    procurementRole: (response.procurementRole || response.role) as User["role"],
    active: true,
    tenantId: response.tenantId,
    tenantName: response.tenantName,
    tenantDomain: response.tenantDomain,
  };
}

export function sessionMetaFromLogin(response: LoginPayload) {
  return {
    tenant: {
      tenantId: response.tenantId,
      tenantName: response.tenantName,
      tenantDomain: response.tenantDomain,
    },
    organizationType: (response.organizationType as OrganizationType) || undefined,
    operatingContext: (response.operatingContext as OperatingContext) || "PROCUREMENT",
    availableContexts: response.availableContexts || [],
    procurementRole: response.procurementRole,
    supplierRole: response.supplierRole,
  };
}
